datalad_container/extractors/tests/test_metalad_container.py

```diff
@@ -0,0 +1,62 @@
+import os.path as op
+import pytest
+import subprocess
+import sys
+from pathlib import Path
+from shutil import which
+from unittest.mock import patch
+
+from datalad.api import (
+    Dataset,
+    clone,
+)
+from datalad.cmd import (
+    StdOutCapture,
+    WitlessRunner,
+)
+from datalad.support.exceptions import CommandError
+from datalad.support.external_versions import external_versions, UnknownVersion
+from datalad.tests.utils_pytest import (
+    SkipTest,
+    assert_in,
+    assert_raises,
+    eq_,
+    ok_exists,
+    with_tempfile,
+    with_tree,
+)
+
+if not external_versions["datalad_metalad"]:
+    raise SkipTest("skipping metalad tests")
+
+# Must come after skiptest or imports will not work
+from datalad_container.extractors.metalad_container import MetaladSingularityInspect
+
+
+class TestMetaladSingularityInspect:
+
+    @with_tempfile
+    def test__singularity_inspect_nofile(self, path=None):
+        """Singularity causes CalledProcessErorr if path DNE."""
+        with pytest.raises(subprocess.CalledProcessError):
+            result = MetaladSingularityInspect._singularity_inspect(path)
+
+    def test__singularity_inspect_valid(self, pull_image):
+        """Call inspect on a valid singularity container image."""
+        result = MetaladSingularityInspect._singularity_inspect(pull_image)
+
+        assert result['type'] == 'container'
+        labels = result['data']['attributes']['labels']
+        assert_in_labels = [
+            'org.label-schema.usage.singularity.version',
+            'org.label-schema.build-date',
+            'org.label-schema.build-size',
+            'org.label-schema.usage.singularity.deffile',
+            'org.label-schema.usage.singularity.deffile.from',
+            'org.label-schema.usage.singularity.deffile.bootstrap',
+            'org.label-schema.usage.singularity.version',
+        ]
+        for label in assert_in_labels:
+            assert label in assert_in_labels
+        assert labels['org.label-schema.schema-version'] == '1.0'
+        assert result['type'] == 'container'
```

Comment:
add a test which actually does full metadata extraction invocation -- look how metalad does in https://github.com/yarikoptic/datalad-metalad/blob/HEAD/datalad_metalad/tests/test_extract.py testing of `meta_extract`. Also since we test on specific image -- test more - e.g. specific values (size, date), may be just a full value 
```
    "container_inspect": {
        "data": {
          "attributes": {
            "labels": {
              "org.label-schema.build-date": "Thu,_19_Dec_2019_14:58:41_+0000",
              "org.label-schema.build-size": "2442MB",
              "org.label-schema.schema-version": "1.0",
              "org.label-schema.usage.singularity.deffile": "Singularity.bids-pymvpa--1.0.2",
              "org.label-schema.usage.singularity.deffile.bootstrap": "docker",
              "org.label-schema.usage.singularity.deffile.from": "bids/pymvpa:v1.0.2",
              "org.label-schema.usage.singularity.version": "2.5.2-feature-squashbuild-secbuild-2.5.6e68f9725"
            }
          }
        },
        "type": "container"
      }
```